src/sprites/setup.py

```diff
@@ -0,0 +1,257 @@
+                     self._exceptions.values(),
+                     [self.default]]:
+            for hitbox in item:
+                hitbox.update(
+                    hitbox.left * factor,
+                    hitbox.top * factor,
+                    hitbox.width * factor,
+                    hitbox.height * factor
+                )
+
+
+class EntityAssets:
+    CHICKEN: EntityAsset
+    COW: EntityAsset
+    RABBIT: EntityAsset
+
+    @classmethod
+    def setup(cls):
+        chicken_hitbox = _Hitbox(pygame.Rect(1, 11, 11, 3))
+        chicken_hitbox.set_direction_exception(
+            Direction.LEFT, pygame.Rect(4, 11, 11, 3)
+        )
+
+        chicken_asset = entity_importer(
+            path=resource_path("images/entities/chicken"),
+            size=16,
+            directions=[Direction.RIGHT],
+            hitbox=chicken_hitbox
+        )
+
+        cls.CHICKEN = EntityAsset(
+            chicken_asset
+        )
+
+        cow_hitbox = _Hitbox(pygame.Rect(6, 25, 16, 4))
+        cow_hitbox.set_direction_exception(
+            Direction.LEFT, pygame.Rect(10, 25, 16, 4)
+        )
+
+        cow_asset = entity_importer(
+            path=resource_path("images/entities/cow"),
+            size=32,
+            directions=[Direction.RIGHT],
+            hitbox=cow_hitbox
+        )
+
+        cls.COW = EntityAsset(
+            cow_asset
+        )
+
+        rabbit_hitbox = _Hitbox(pygame.Rect(18, 26, 12, 6))
+        rabbit_hitbox.set_direction_exception(
+            Direction.LEFT, pygame.Rect(20, 26, 8, 6)
+        )
+        rabbit_hitbox.set_direction_exception(
+            Direction.RIGHT, pygame.Rect(20, 26, 8, 6)
+        )
+        rabbit_hitbox.set_exception(
+            EntityState.AXE, Direction.LEFT, pygame.Rect(24, 26, 8, 6)
+        )
+        rabbit_hitbox.set_exception(
+            EntityState.AXE, Direction.RIGHT, pygame.Rect(16, 26, 8, 6)
+        )
+
+        rabbit_asset = entity_importer(
+            path=resource_path("images/characters/rabbit"),
+            size=CHAR_TILE_SIZE,
+            directions=[Direction.DOWN, Direction.UP, Direction.LEFT],
+            hitbox=rabbit_hitbox
+        )
+
+        cls.RABBIT = EntityAsset(
+            rabbit_asset
+        )
+
+
+def state_importer(
+        path: str, size: int, state: EntityState, directions: list[Direction],
+        hitbox: _Hitbox
+) -> dict[Direction, _AniFrames]:
+
+    directions_dict = {}
+    full_path = os.path.join(path)
+    surf = pygame.image.load(full_path).convert_alpha()
+
+    for row, direction in enumerate(directions):
+        frames = []
+
+        for col in range(surf.get_width() // size):
+            cutout_surf = pygame.Surface((size, size), pygame.SRCALPHA)
+            cutout_rect = pygame.Rect(
+                col * size,
+                row * size,
+                size,
+                size
+            )
+
+            cutout_surf.blit(surf, (0, 0), cutout_rect)
```

Comment:
Actually, this could also be simplified by just subsurfacing the original surface here instead of creating a new surface and copying the pixel data on it

Reply:
Done! 😄